src/Workspaces/Remote/Core/Services/SolutionService.cs

```diff
@@ -69,7 +71,7 @@ private OptionSet MergeOptions(OptionSet workspaceOptions, OptionSet userOptions
             var solutionChecksumObject = await RoslynServices.AssetService.GetAssetAsync<SolutionChecksumObject>(solutionChecksum, cancellationToken).ConfigureAwait(false);
 
             // TODO: Make these to do work concurrently
-            var workspace = new AdhocWorkspace(RoslynServices.HostServices);
+            var workspace = new AdhocWorkspace(RoslynServices.HostServices, workspaceKind: WorkspaceKind);
```

Comment:
we do this so that we pick right workspace service when both workspace runs in same process for testing purpose. at the end, I will probably create actual RemoteHostWorkspace rather than using AdhocWorkspace. but for now, I am just giving it its own workspace kind